packages/headless/src/controllers/insight/generated-answer/headless-insight-generated-answer.ts

```diff
@@ -34,8 +35,8 @@ export function buildGeneratedAnswer(
   props: GeneratedAnswerProps = {}
 ): GeneratedAnswer {
   const {dispatch} = engine;
-  const controller = buildCoreGeneratedAnswer(
-    engine,
+  const controller = buildSearchAPIGeneratedAnswer(
+    engine as unknown as SearchEngine, // TODO: types
```

Comment:
This cast is disgracious. a better solution seems necessary.

Reply:
Cannot find any better solution than to cast here.

Reply:
are we eventually going to remove the buildSearchApiGeneratedAnswer and only use the answer api ?

Reply:
This is a good question. I think the end goal is yes. But lets admit that this may take quite a while since we will need all users leave the current flow. Which probably mean all our customer migrate to V3, it may (and likely will) take quite a while.

Reply:
With the suggestions I made above I think you should be able to get rid of the cast.

Reply:
This narrowing is way better than this ugly cast. 

thanks a lot.